test/rbi/parser_test.rb

```diff
@@ -88,6 +88,53 @@ def test_parse_constants
       assert_equal(rbi, tree.string)
     end
 
+    def test_parse_constants_with_heredoc
+      rbi = <<~RBI
+        A = <<-EOF
```

Comment:
I am not sure if these examples are exhaustive enough. For example, a common pattern in the CRuby codebase is to use something like:
```ruby
Foo = #{<<~begin}#{<<~end}
begin
  foo
  foo
end
```
like in here: https://github.com/ruby/ruby/blob/eafcdc153560fd391dabd60705cb88e3f72d7b47/test/ruby/test_file_exhaustive.rb#L760-L773 and this is the main reason why heredocs aren't necessarily contiguous in their begin-end range, and why the tag that represents the heredoc is conceptually separate from the body of it.

I think we need to test to see how multiple heredocs on the same line will interact with this.